libs/kotaemon/kotaemon/indices/vectorindex.py

```diff
@@ -69,6 +70,13 @@ def run(self, text: str | list[str] | Document | list[Document]):
         if self.doc_store:
             print("Adding documents to doc store")
             self.doc_store.add(input_)
+        # save the text content into markdown format
+        if self.cache_dir is not None:
+            file_name, _ = os.path.splitext(input_[0].metadata['file_name'])
```

Comment:
Now Python has a more advanced `pathlib` module, I think we should avoid using `os.path.xxx` and using the pathlib's equivalent methods when possible.